File: Backend-Project/src/main/java/com/example/demo/entities/Formateur.java
package com.example.demo.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@EqualsAndHashCode(callSuper = false)
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
@DiscriminatorValue("FORM")
public class Formateur extends Utilisateur{
    @ManyToMany(mappedBy = "formateurs",fetch = FetchType.EAGER)
    private List<Competence> competences = new ArrayList<>();

    @OneToMany(mappedBy = "formateur")
    private List<Validation> validations = new ArrayList<>();
}

File: argon-design-system-angular/src/entities/formateur.ts
import { Competence } from './competence';
import { Validation } from './validation';
import { Utilisateur } from './utilisateur';

export interface Formateur extends Utilisateur{
    competences : Competence[];
    validations : Validation[];
    id: number;
    prenom: string;
    nom: string;
    password: string;
    email: string;
    telephone: string;
    role: string;

}